webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -789,75 +862,91 @@ class PolicyWriterServiceImpl(
                               val replace = (s: String) => StringUtils.replace(s, Policy.TAG_OF_RUDDER_MULTI_POLICY, id.getRudderUniqueId)
                               (replace(filled), replace(templateInfo.destination))
                           }
-      _                <- tryo { FileUtils.writeStringToFile(new File(outPath, dest), replaced, StandardCharsets.UTF_8) } ?~!
-                            s"Bad format in Technique ${templateInfo.id.toString} (file: ${templateInfo.destination})"
+      _                <- File(outPath, dest).createParentsAndWrite(replaced).chainError(
+                              s"Bad format in Technique ${templateInfo.id.toString} (file: ${templateInfo.destination})"
+                          )
+      t2               <- currentTimeNanos
+      _                <- writeTimer.writeTemplate.update(_ + t2 - t1)
     } yield {
       outPath
     }
   }
 
   /**
    * Move the machine promises folder  to the backup folder
-   * @param machineFolder
+   * @param nodeFolder
    * @param backupFolder
    */
-  private[this] def backupNodeFolder(nodeFolder: String, backupFolder: String): Unit = {
-    val src = new File(nodeFolder)
-    if (src.isDirectory()) {
-      val dest = new File(backupFolder)
-      if (dest.isDirectory) {
-        // force deletion of previous backuup
-        FileUtils.deleteDirectory(dest)
+  private[this] def backupNodeFolder(nodeFolder: String, backupFolder: String): IOResult[Unit] = {
+    IOResult.effect {
+      val src = File(nodeFolder)
+      if (src.isDirectory()) {
+        val dest = File(backupFolder)
+        if (dest.isDirectory) {
+          // force deletion of previous backuup
+          dest.delete(false, File.LinkOptions.noFollow)
+        }
+        PolicyLogger.trace(s"Backup old '${nodeFolder}' into ${backupFolder}")
+        dest.parent.createDirectoryIfNotExists(true)
+        src.moveTo(dest)(File.CopyOptions.atomically)
```

Comment:
ok, if we do that, we need to enforce that users are using the same FS for backup and for policies, otherswise it raises a AtomicNotSupportedException

Reply:
and do you know how much of the improvement comes from the atomically, compared to switching to betterFiles ?

Reply:
I've added a test that choose between `atomic` or not, and write a warning if it's not supported. But the more I test, the less I believe `atomic` is the reason of the performance boost (if my test always return `default`, I get the same result - but perhaps linux is smart and it does an atomic move by default if no option is given ?)

Reply:
Linux is smart. The change is in the cache FS first, then is commited when it's full

Reply:
Ok :)